aptos-move/block-executor/src/errors.rs

```diff
@@ -13,28 +14,29 @@ pub enum IntentionalFallbackToSequential {
     /// TODO: (short-mid term) relax the limitation, and (mid-long term) provide proper multi-versioning
     /// for code (like data) for the cache.
     ModulePathReadWrite,
-    /// We defensively check certain resource group related invariant violations.
-    ResourceGroupError(String),
+    /// We defensively check resource group serialization error in the commit phase.
+    /// TODO: should trigger invariant violation in the transaction itself.
+    ResourceGroupSerializationError(String),
 }
 
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub enum Error<E> {
+pub enum BlockError<E> {
     FallbackToSequential(PanicOr<IntentionalFallbackToSequential>),
```

Comment:
nit: This is more a description of the action that needs to be performed - vs the actual error. Maybe rename?
`ConccurencyConflictDetected` or something like that?

Reply:
It's not just concurrency conflicts, I'll let @igor-aptos opine here but we can do it separately.

Reply:
I think the information on the error goes into IntentionalFallbackToSequential. 

and the main distinction here is the intention - do we FallbackToSequential or have another FatalVMError